lint-rules.md

```diff
@@ -0,0 +1,1280 @@
+
+If bracketed content (things between `()` or `[]`) must be split onto multiple lines, a newline should follow the opening bracket, the contents should be indented an additional level, then the closing bracket should be de-indented to match the indentation of the opening bracket.
+
+If you are line splitting an expression on an infix operator, the operator and at least the beginning of the RHS operand should be on the continued line. (i.e. an operator should _not_ be on a line by itself.)
+
+If you are using the `if...then...else...` construct as part of your expression and it needs to be line split, the entire construct should be wrapped in parentheses (`()`). The opening parenthesis should be immediately followed by a newline. `if`, `then`, and `else` should all start a line one more level of indentation than the wrapping paratheses. The closing parenthesis should be on the same level of indentation as the opening parenthesis.
+
+If you are using the `if...then...else...` construct on one line, it does not need to be wrapped in parentheses. However, if any of the 3 clauses are more complex than a single identifier, they should be wrapped in parentheses.
+
+Sometimes a developer will choose to line split an expression despite it being able to all fit on one line that is <=90 characters wide. That is perfectly acceptable, though you may notice in the below examples the single line form can be more readable. There is "wiggle" room allowed by the above rules. This is intentional, and allows developers to choose a more compact or a more spaced out expression.
+
+**Group**: `spacing`
+
+#### Example
+
+Complex example that fits on one line:
+
+```wdl
+Int complex_value = w - x + (y * (z / (foo % bar)))
+```
+
+Same example with as much line splitting as permissible:
+
+```wdl
+Int complex_value
+    = w
+    - x
+    + (
+        y
+        * (
+            z
+            / (
+                foo
+                % bar
+            )
+        )
+    )
+```
+
+Another complex example written as compactly as permissible:
+
+```wdl
+Boolean complicated_logic = (
+    if !(
+        a && b || c && (!d || !e)
+        == (
+            if foobar
+            then gak
+            else caz
+        )
+    )
+    then "wow"
+    else "WOWOWOW"
+)
+```
+
+The same expression with as many newlines inserted as permissible:
+
+```wdl
+Boolean complicated_logic
+    = (
+        if
+            !(
+                a
+                && b
+                || c
+                && (
+                    !d
+                    || !e
+                )
+                == (
+                    if
+                        foobar
+                    then
+                        gak
+                    else
+                        caz
+                )
+            )
+        then
+            "wow"
+        else
+            "WOWOWOW"
+    )
+```
+
+### `input_spacing`
```

Comment:
- Typo: "dcoument" to "document"
- Disagree with this lint generally; I prefer having `input:` on its own line, which helps emphasize that this is a call to a task, and also allows for WDL to evolve to allow other types of blocks to pass to the task (maybe `meta` info in the future? something else?) so that this lint would "scale" more naturally